### PR TITLE
Divyanshu | Test | Implementation of Zimbabwe VAT in GIDDH

### PR DESCRIPTION
**Pull request by dvCodeWorld**
Zimbabwe - https://app.clickup.com/t/86cuy39kd

### CODE REVIEW

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 1]
```diff
@@ -43,18 +43,21 @@
                                     <div *ngIf="taxes?.length > 1" class="trn-width">
                                         <dropdown-field [label]="commonLocaleData?.app_trn" [options]="taxes"
                                             (selectedOption)="getVatReport($event)"
-                                            [placeholder]="commonLocaleData?.select_trn" [defaultValue]="taxes[0]?.label"
-                                            [required]="true"></dropdown-field>
+                                            [placeholder]="commonLocaleData?.select_trn"
+                                            [defaultValue]="taxes[0]?.label" [required]="true"></dropdown-field>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]: format code

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 2]
```diff
@@ -288,75 +290,182 @@ <h1>{{ localeData?.no_vrn_available }}</h1>
...
-                                            class="table basic mr-t2 on-mobile-view vat-report-table">
-                                            <ng-container matColumnDef="number">
-                                                <td mat-cell *matCellDef="let element" width="5%">
-                                                    {{ localeData?.box }} {{ element.order }}
+                                            class="table basic mr-t2 on-mobile-view vat-report-table zimbabwe-vat-report-table">
+                                            <ng-container matColumnDef="name">
+                                                <th class="d-none" mat-header-cell *matHeaderCellDef></th>
+                                                <td mat-cell *matCellDef="let element" [attr.colspan]="(i === 1 && element?.order === 9) ? 6 : 1"
+                                                    [ngClass]="{'bg-grey font-weight-bold':(i === 0 && element?.order === 8) || (i === 1 && element?.order === 9)}">
+                                                    <span>{{ element.description }}</span>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                                    <span>{{ element?.description }}</span>
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 3]
```diff
@@ -288,75 +290,182 @@ <h1>{{ localeData?.no_vrn_available }}</h1>
...
+                                                    <ng-container *ngIf="!((i === 0 && element?.order === 8) || (i === 1 && element?.order === 9))">
+                                                        {{ element?.matCode }}
+                                                    </ng-container>
+                                                    <ng-container *ngIf="(i === 0 && element?.order === 8) || (i === 1 && element?.order === 9)">
+                                                        {{ element?.value }}
+                                                    </ng-container>
+                                                </td>
+                                                <td mat-footer-cell *matFooterCellDef colspan="3" class="text-right">
+                                                    <span *ngIf="i === 0">
+                                                        &#40;A&#41;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]: Confirm with @ravindersinghdang sir

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 4]
```diff
@@ -288,75 +290,182 @@ <h1>{{ localeData?.no_vrn_available }}</h1>
...
-                                                    {{ report?.currency?.symbol + ((element.outputTax.amount + (0.0 +  element.outputTax.decimal)) | giddhCurrency) }}
+                                            <ng-container matColumnDef="vos-amount">
+                                                <th mat-header-cell *matHeaderCellDef>
+                                                    <p class="inner-table-heading text-right font-weight-bold">
+                                                        {{ vatReportCurrencyMap[0] }}
+                                                    </p>
+                                                </th>
+                                                <td mat-cell *matCellDef="let element" class="text-right"
+                                                    [ngClass]="{'d-none': (i === 0 && element?.order === 8) || (i === 1 && element?.order === 9)}">
+                                                    {{ vatReportCurrencySymbol + element.valueOfSupply?.amount }}
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                                    {{ vatReportCurrencySymbol + element?.valueOfSupply?.amount }}
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 5]
```diff
@@ -288,75 +290,182 @@ <h1>{{ localeData?.no_vrn_available }}</h1>
...
-                                            <tr mat-row *matRowDef="let row; columns: ukDisplayedColumns"></tr>
+                                            <ng-container matColumnDef="vos-decimal">
+                                                <th mat-header-cell *matHeaderCellDef>
+                                                    <p class="inner-table-heading text-right font-weight-bold">
+                                                        {{ vatReportCurrencyMap[1] }}
+                                                    </p>
+                                                </th>
+                                                <td mat-cell *matCellDef="let element" class="text-right"
+                                                    [ngClass]="{'d-none': (i === 0 && element?.order === 8) || (i === 1 && element?.order === 9)}">
+                                                    {{ vatReportCurrencySymbol + element.valueOfSupply?.decimal }}
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                                    {{ vatReportCurrencySymbol + element?.valueOfSupply?.decimal }}
```

File: apps/web-giddh/src/assets/css/global/vat-report.scss [hunk 6]
```diff
@@ -1,7 +1,45 @@
...
+                        padding: 5px 10px !important;
+                    }
+
+                    th,td {
+                        border:  1px solid var(--color-table-primary-content-bar-border) !important;
+        
+                        &[colspan="2"]{
+                            text-align: center;
+                        }
+                        &.d-none {
```
Comment (nisha-walkover) [inline apps/web-giddh/src/assets/css/global/vat-report.scss:18]: why add display: none;  css 
when already using bootstrap class d-none

File: apps/web-giddh/src/assets/css/global/vat-report.scss [hunk 7]
```diff
@@ -1,7 +1,45 @@
...
+                        }
+                    }
+                }
+            }
+        }
+        
         tfoot {
             background-color: var(--color-wild-sand);
+            tr {
+                td {
```
Comment (nisha-walkover) [inline apps/web-giddh/src/assets/css/global/vat-report.scss]: for font-weight 700 
use bootstrap4 class font-weight-bold
bootstrap5 fw-bold

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 8]
```diff
@@ -288,75 +290,182 @@ <h1>{{ localeData?.no_vrn_available }}</h1>
...
-                                                table-border-top
-                                            ">
-                                            <ng-container matColumnDef="number">
+                                        <table mat-table [dataSource]="report.sections"
+                                            class="table basic mr-t2 on-mobile-view vat-report-table zimbabwe-vat-report-table">
+                                            <ng-container matColumnDef="name">
                                                 <th mat-header-cell *matHeaderCellDef></th>
                                                 <td mat-cell *matCellDef="let element">
-                                                    {{ localeData?.box }} {{ element.order }}
+                                                    <span [ngClass]="{'font-weight-bold': showTotalOutputTaxIn.includes(element?.order)}">{{
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                                    <span [ngClass]="{'font-weight-bold': showTotalOutputTaxIn?.includes(element?.order)}">{{
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts [hunk 9]
```diff
@@ -123,7 +131,11 @@ export class VatReportComponent implements OnInit, OnDestroy {
         { label: 'EUR', value: 'EUR', additional: { symbol: '€' } }
     ];
     /** Holds Current Currency Code for Zimbabwe report */
-    public vatReportCurrency: 'BWP' | 'USD' | 'GBP' | 'INR' | 'EUR' = this.vatReportCurrencyList[0].value;
+    public vatReportCurrencyCode: 'BWP' | 'USD' | 'GBP' | 'INR' | 'EUR' = this.vatReportCurrencyList[0].value;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.ts:134]:
```suggestion
    public vatReportCurrencyCode: 'BWP' | 'USD' | 'GBP' | 'INR' | 'EUR' = this.vatReportCurrencyList[0]?.value;
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts [hunk 10]
```diff
@@ -123,7 +131,11 @@ export class VatReportComponent implements OnInit, OnDestroy {
...
+    /** Holds Current Currency Symbol for Zimbabwe report */
+    public vatReportCurrencySymbol: string = this.vatReportCurrencyList[0].additional.symbol;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.ts:136]:
```suggestion
    public vatReportCurrencySymbol: string = this.vatReportCurrencyList[0]?.additional?.symbol;
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts [hunk 11]
```diff
@@ -280,6 +292,12 @@ export class VatReportComponent implements OnInit, OnDestroy {
                         this.isLoading = false;
                         if (res.status === 'success') {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.ts]:
```suggestion
                        if (res && res.status === 'success' && res.body) {
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts [hunk 12]
```diff
@@ -280,6 +292,12 @@ export class VatReportComponent implements OnInit, OnDestroy {
...
                             this.vatReport = res.body?.sections;
+                            if(this.activeCompany?.countryV2?.alpha2CountryCode === 'ZW') {
+                                this.vatReportCurrencyCode = res?.body?.currency?.code;
+                                this.vatReportCurrencySymbol = this.vatReportCurrencyList.filter( item => item.label === res?.body?.currency?.code).map(item => item.additional.symbol).join();
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.ts]: remove all nulity check for res.body

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts [hunk 13]
```diff
@@ -516,7 +534,7 @@ export class VatReportComponent implements OnInit, OnDestroy {
      */
     public onCurrencyChange(event: any): void {
         if (event) {
-            this.vatReportCurrency = event?.value;
+            this.vatReportCurrencyCode = event?.value;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.ts]: remove nulity check for event.value already check in if condition

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/assets/css/global/vat-report.scss (hunk 6 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/assets/css/global/vat-report.scss:18]: to over ride mat table display property, without using more descriptive information I unable to apply display none property

File: apps/web-giddh/src/assets/css/global/vat-report.scss (hunk 7 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/assets/css/global/vat-report.scss]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 1 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 2 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 4 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 5 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 8 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts (hunk 9 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.ts:134]: No need, It always get value.

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts (hunk 10 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.ts:136]: No need, It always get additional and symbol.

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts (hunk 11 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.ts]: Done

if (res && res?.status === 'success' && res?.body)

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts (hunk 12 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.ts]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts (hunk 13 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.ts]: I checked only event, but as per your suggestion I update my code as 

 if (event?.value) {
            this.vatReportCurrencyCode = event.value;
            this.getVatReport();
        }

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 3 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: ?

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts (hunk 13 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/vat-report/vat-report.component.ts]: I said remove only nulity check in event.value already you are check event in if condition

File: apps/web-giddh/src/assets/css/global/vat-report.scss [hunk 14]
```diff
@@ -35,11 +35,11 @@
...
         tfoot {
             background-color: var(--color-wild-sand);
-            tr {
-                td {
-                    font-weight: 700 !important;
-                }
-            }
+            // tr {
+            //     td {
+            //         font-weight: 700 !important;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/assets/css/global/vat-report.scss]: remove commented code

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/assets/css/global/vat-report.scss (hunk 14 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/assets/css/global/vat-report.scss]: Done

**Review by dilpreetsinghofficial — APPROVED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**